### DISCUSSION TITLE
Changing Font, text, and background colors. NVS problems.

### DISCUSSION THREAD

**Discussion by lowerclasswarfare** [toverainc/willow#350] [Show and tell]
EDIT: By the way, I LOVE this project. It's really great so far. Thank you for all your hard work.

So I didn't know if I should bring this up as an issue ticket so I just decided to post it here.

I wanted to change the default font because I didn't like it (no offense devs). So I decided to set up the development environment, got all that fun stuff running, jumped in the code and figured out how to convert ttf fonts into a bin file, then changed some hex values for text and bg color. Everything flashed fine but there was some issue with the nvs partition which I figured out by searching GitHub. Long story short - nvs partition gets messed up from erase-flash when you use it from the dev environment. I had to reflash with the quick start guide, THEN flash my custom image. It didn't matter how many times I tried to flash a custom nvs partition. Still wouldn't work. Anyways here's some screenshots of my ESP box now.

![willow](https://github.com/toverainc/willow/assets/51834266/d5571483-b471-41b9-ab0e-19397cde76d5)
![20231218_233601](https://github.com/toverainc/willow/assets/51834266/80ddb8f3-a27b-4d38-beb1-4ff17c147480)
![20231218_233705](https://github.com/toverainc/willow/assets/51834266/2cc09f2f-06ef-47b9-ace1-e94797e8e2ad)
![20231218_233746](https://github.com/toverainc/willow/assets/51834266/c43cc038-db53-4606-9906-9674971de3a8)

**Comment by kristiankielhofner** [comment]
Cool! Nice work. I assume you found `./utils.sh convert-font`? No worries on our default font, we had a community member design our "branding" and that was the font they selected. It's consistent across the web, display, etc.

For the NVS issue you describe - do you have your WiFi + WAS settings set correctly under "Willow Configuration" from the build menu?

**Reply by lowerclasswarfare** [reply]
Oh there's a convert-font option? Woops 😂
I went into ui.c and did all that by hand. I used an online font converter. I think it was on openhab? I think there's one on espressifs site.
Yeah I can see why you would want a consistent theme across everything! I just like to tinker with stuff.

As far as I understand my Wifi and WAS settings are valid under the menu yeah. Watching the serial console it just kept spitting out "Disconnected from AP retrying."

**Reply by kristiankielhofner** [reply]
> As far as I understand my Wifi and WAS settings are valid under the menu yeah. Watching the serial console it just kept spitting out "Disconnected from AP retrying."

This is very strange and should not be happening. Can you provide your command sequence from build to flash?

**Reply by stintel** [reply]
If the NVS partition was somehow not flashed, the Wi-Fi should not even try to connect. The log suggests that Willow is unable to connect using the credentials found in the NVS partition. I would double check the credentials and keep in mind SSID is case sensitive. IIRC we had several users who were not aware of that.

Maybe also double check in the sdkconfig file itself, there might be a stray whitespace.

**Reply by lowerclasswarfare** [reply]
@kristiankielhofner Yeah I'll try to reproduce the steps here. I followed the development section on the willow homepage and I'm running Manjaro if that helps.

```shell
./utils.sh build-docker
./utils.sh docker
./utils.sh install
./utils.sh config

##then in menuconfig Audio HAL ---> Audio Board ---> ESP32-S3-BOX-3
##and in Willow Configuration I enter my WiFi + SSID + WAS server details
##edit ui.c in main directory

./utils.sh build
exit
export PORT=/dev/ttyACM0
./utils.sh flash
#complained about having to run erase-flash
./utils.sh erase-flash
./utils.sh flash

```
And that's it really. I didn't do much different other than trying to destroy the environment after I was having NVS issues, and then re-building and flashing.


@stintel Essentially what happened was I had already flashed the box with Willow following the regular guide. It worked just fine out of the box. After messing with it for awhile I decided to play with the source code and set up my environment, and when I got to the point of flashing it said I had to "run erase-flash first" so I did, then flashed my built image which THEN complained about not having NVS settings. I had double and triple checked my ssid (which is all lowercase) and pass so I found a thread on here explaining how to generate and flash a partition to 0x9000. Tried that, it flashed successfully but that's when I started having AP disconnection issues. I will definitely check the config for stray whitespace which could make sense.

**Reply by stintel** [reply]
Looking at the code in utils.sh again, you need to run:

```
./utils.sh dist
./utils.sh flash-dist
```
The other flash commands will not include the NVS partition. Apologies for the wrong info, we haven't really used utils.sh for flashing as we're doing mostly via Web Flash or WAS OTA now.
